### PR TITLE
Adding Pairs prices

### PR DESCRIPTION
**Pull request by GabrielCamba**
Added token1 and token0 prices to pairs, daily and hourly ones as requested by coingecko

deployed at: https://thegraph.com/hosted-service/subgraph/gabrielcamba/cow-staging?version=pending

### CODE REVIEW

**Review by anxolin — APPROVED** [review]

File: src/modules/pairs.ts [hunk 1]
```diff
@@ -4,140 +4,167 @@ import { ZERO_BD, ZERO_BI } from "../utils/constants";
...
+    ) {
+      this.token = _token
+      this.volume = _volume
+      this.price = _price
+    }
+  }
+
+  export function createOrUpdatePair(timestamp: BigInt, buyTokenId: string, sellTokenId: string, buyAmount: BigInt, sellAmount: BigInt,
+    sellAmountEth: BigDecimal, sellAmountUsd: BigDecimal, buyTokenPrice: BigDecimal, sellTokenPrice: BigDecimal): void {
+    let canonicalMarket = getCanonicalMarket(buyTokenId, sellTokenId, buyAmount, sellAmount, buyTokenPrice, sellTokenPrice)
```
Comment (anxolin) [inline src/modules/pairs.ts:25]: @GabrielCamba im not sure if i understand here, but if i follow the logic correctly. This `sellTokenPrice` is in reality the USD price.

https://github.com/cowprotocol/subgraph/blob/2467789a7347c8cd8ee52d23325fff6c3cc51c37/src/modules/trades.ts#L46

image


1st concern, is our naming convention. I feel that if the price is USD we should say it in the name. 

2nd concern is that if I understand this code, you are using this price later https://github.com/cowprotocol/subgraph/pull/24/files#diff-26dee9f91cd639f585c0d6e3f6a0d634300c0a86c6ec48b913c49d699ff5b43fR30

https://github.com/cowprotocol/subgraph/blob/2467789a7347c8cd8ee52d23325fff6c3cc51c37/src/modules/pairs.ts#L41

Is it possible you use the "Usd" price as if it was the price of the token expressed in the other token of the Pair?

File: src/modules/pairs.ts [hunk 2]
```diff
@@ -4,140 +4,167 @@ import { ZERO_BD, ZERO_BI } from "../utils/constants";
...
+
+    let buyTokenProps = new TokenProps(buyTokenId, buyAmount, buyTokenPrice)
+    let sellTokenProps = new TokenProps(sellTokenId, sellAmount, sellTokenPrice)
+
+    if (buyTokenNumber.lt(sellTokenNumber)) {
+      value.set("token0", buyTokenProps)
+      value.set("token1", sellTokenProps)
+    } else {
+      value.set("token0", sellTokenProps)
+      value.set("token1", buyTokenProps)
```
[the remark this answers is not in the document]
Comment (GabrielCamba) [reply src/modules/pairs.ts:60]: Yeah I noticed this when I had to add a new value (with a new type) there :man_facepalming:

File: src/modules/pairs.ts (hunk 1 above)
Comment (GabrielCamba) [reply src/modules/pairs.ts:25]: > Is it possible you use the "Usd" price as if it was the price of the token expressed in the other token of the Pair?

I will work on this.

Comment (anxolin) [reply src/modules/pairs.ts]: Is this topic still open?

Comment (GabrielCamba) [reply src/modules/pairs.ts]: I think this is already fixed.

**Pull request merged**